src/test/java/reactor/kafka/receiver/KafkaReceiverTest.java

```diff
@@ -1190,6 +1190,29 @@ public void abortTransaction() throws Exception {
         checkConsumedMessages(count, count * 3);
     }
 
+    @Test
+    public void userPause() throws InterruptedException {
```

Comment:
does the test really cover externally paused topic partitions? ie. would there be a way to have 2 partitions paused and verify that the `PollEvent` distinguishes between the two more explicitly?

Reply:
Yes; more test coverage is needed; I originally wrote this test just to demonstrate that the paused partition was incorrectly resumed. 

The notion of user Vs. our pauses only came to me while fixing the bug, but I didn't enhance the test to cover that use case.